src/components/search/UseSearch.jsx

```diff
@@ -31,16 +33,26 @@ const useSearch = () => {
     } catch (error) {
       console.error('Failed to fetch classes:', error.message);
     } finally {
-      setLoading(false);
+      setTimeout(() => {
```

Comment:
What's the purpose of this `setTimeout`?

Reply:
the rendering looked smoother but can remove if it is an issue

Reply:
we can leave this for now.